sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs

```diff
@@ -1457,6 +1458,44 @@ public async Task GetPropertiesAsync_BlobSAS()
 
         }
 
+        [Test]
```

Comment:
This test passes even without the change to AppendProperties, since we still create a new Uri which encodes the data for us. 
I still think the fix makes sense so that the query string is properly encoded independent of the Uri, especially since we do this for the dates already. 
Can we make the test validate that the return data from ToSasQueryParameters is encoded? 
We can also validate that the data doesn't end up getting double encoded.

Reply:
Added both checks.

Reply:
Looks good